src/amg/azext_amg/migrate.py

```diff
@@ -0,0 +1,331 @@
+        all_annotations, restore_url, restore_headers, dry_run, overwrite)
+
+    dry_run_status = "to be " if dry_run else ""
+    output = [
+        (Style.IMPORTANT, f"\n\nDry run: {dry_run}\n"),
+        (Style.IMPORTANT, f"Overwrite dashboards, folders, and library panels: {overwrite}\n"),
+    ]
+
+    append_summary(output, dry_run_status + "created", "Datasources", datasources_created_summary,
+                   "\nDatasources were created, credentials ARE NOT migrated.")
+    append_summary(output, dry_run_status + "remapped", "Datasources", datasources_remapped_summary)
+    append_summary(output, dry_run_status + "created", "Folders", folders_created_summary)
+    append_summary(output, dry_run_status + "overwritten", "Folders", folders_overwrote_summary)
+
+    append_nested_summary(output, dry_run_status + "created", "Library panels", library_panels_created_summary)
+    append_nested_summary(output, dry_run_status + "overwritten", "Library panels", library_panels_overwrote_summary)
+    append_nested_summary(output, dry_run_status + "created", "Dashboards", dashboards_created_summary)
+    append_nested_summary(output, dry_run_status + "overwritten", "Dashboards", dashboards_overwrote_summary)
+    append_nested_summary(output, dry_run_status + "created", "Snapshots", snapshots_synced_summary)
+    append_nested_summary(output, dry_run_status + "overwritten", "Snapshots", snapshots_overwrote_summary)
+
+    append_annotations_summary(output, dry_run_status + "created", "Annotations", annotations_synced_summary)
+    append_annotations_summary(output, dry_run_status + "overwritten", "Annotations", annotations_overwrote_summary)
+
+    print_styled_text(output)
+
+
+def append_summary(output, status, summary_type, summary_data, important_message=None):
+    if len(summary_data) > 0:
+        output.append((Style.SUCCESS, f"\n\n{summary_type} {status}:"))
+        if important_message:
+            output.append((Style.IMPORTANT, important_message))
+        output.append((Style.PRIMARY, "\n    " + "\n    ".join(summary_data)))
+
+
+def append_nested_summary(output, status, summary_type, summary_data):
+    output.append((Style.SUCCESS, f"\n\n{summary_type} {status}:"))
+    for folder, items in summary_data.items():
+        output.append((Style.PRIMARY, f"\n    {folder}/\n        "))
+        output.append((Style.SECONDARY, "\n        ".join(items)))
+
+
+def append_annotations_summary(output, status, summary_type, annotations_summary, max_text_length=50):
+    output.append((Style.SUCCESS, f"\n\n{summary_type} {status}:"))
+    if len(annotations_summary) > 0:
+        postprocessed_summary = [(a, b[:max_text_length] + "...") if len(b) >
+                                 max_text_length else (a, b) for a, b in annotations_summary]
+        output.append((Style.PRIMARY, "\n    " +
+                       "\n    ".join([f"id: {a}, text: {b}" for a, b in postprocessed_summary])))
```

Comment:
what's the distinction between these 3 different summary helpers? Is it possible to simplify it into 1?

Reply:
There might be a way to simplify into one, but I think it will make the code messier. They are used to show:
![image](https://github.com/user-attachments/assets/96545a8a-7c29-4f75-8be8-7faa3c8a4abc)

So append_summary would take in a simple list, while append_nested_summary takes in a dictionary, the reason why is because we need things to go under folders so we also want to show those. The simple list are for components that don't need folders. 

append_annotations_summary takes in a list, but each lsit has id & text, for annotation to show more information about the annotation